node/tests/TestUtilities.ts

```diff
@@ -51,6 +51,36 @@ export function getFirstResult(
     return Object.values(res).at(0);
 }
 
+/**
+ * Compare two maps by converting them to JSON strings and checking for equality, including property order.
+ *
+ * @param map - The first map to compare.
+ * @param map2 - The second map to compare.
+ * @returns Returns nothing, but throws an error if the maps are not equal.
+ * @remarks This function is used to compare maps, including their property order.
+ * Direct comparison with `expect(map).toEqual(map2)` might ignore the order of properties,
+ * whereas this function considers property order in the comparison by converting the maps to JSON strings.
+ * This ensures a stricter comparison that takes property order into account.
+ *
+ * @example
+ * ```typescript
+ * const mapA = { name: 'John', age: 30 };
+ * const mapB = { age: 30, name: 'John' };
+ *
+ * // Direct comparison will pass because it ignores property order
+ * expect(mapA).toEqual(mapB); // This will pass
+ *
+ * // Correct comparison using compareMaps function
+ * compareMaps(mapA, mapB); // This will fail due to different property order
+ * ```
+ */
+export function compareMaps(
+    map: Record<string | number | symbol, unknown>,
+    map2: Record<string | number | symbol, unknown>,
+) {
+    expect(JSON.stringify(map)).toEqual(JSON.stringify(map2));
```

Comment:
return true for equal and false if not. then do the expect true or false inside the tests. it is more readable from the function name and from reading the test